### ISSUE TITLE
Custom Provider: data.Get("foo") returns empty list

### ISSUE THREAD

**Issue by edulop91** [hashicorp/terraform#23054]
### Terraform Version

```
0.12.10
```

### Terraform Configuration Files

```hcl
resource "snowflake_role_grants" "this_does_not_work" {
  role_name = snowflake_role.foo.name
  roles     = ["roleA"]
  users = [
    "userA",
    "userB",
    snowflake_user.user["userC"].name,
    snowflake_user.user["user_not_exists"].name, // does not work even after removing this
  ]
}
```

```hcl
resource "snowflake_role_grants" "this_works" {
  role_name = snowflake_role.foo.name
  roles     = ["roleA"]
  users = [
    "userA",
    "userB",
    "userC",
  ]
}
```
The provider schema:
```go

		Schema: map[string]*schema.Schema{
			"role_name": &schema.Schema{
				Type:        schema.TypeString,
				Elem:        &schema.Schema{Type: schema.TypeString},
				Required:    true,
				Description: "The name of the role we are granting.",
				ValidateFunc: func(val interface{}, key string) ([]string, []error) {
					return snowflake.ValidateIdentifier(val)
				},
			},
			"roles": &schema.Schema{
				Type:        schema.TypeSet,
				Elem:        &schema.Schema{Type: schema.TypeString},
				Optional:    true,
				Description: "Grants role to this specified role.",
			},
			"users": &schema.Schema{
				Type:        schema.TypeSet,
				Elem:        &schema.Schema{Type: schema.TypeString},
				Optional:    true,
				Description: "Grants role to this specified user.",
			},
```

### Expected Behavior

This happens when developing a provider. I have noticed that during the Update part of the resource cycle when I run `data.Get("users")`. 
```hcl
resource "snowflake_role_grants" "this_does_not_work" {
  role_name = snowflake_role.foo.name
  roles     = ["roleA"]
  users = [
    "userA",
    "userB",
    snowflake_user.user["userC"].name
  ]
}
```
I would expect to get the following list:
```
[userA, userB, userC]
```
But instead I always get an empty list.

If I change the hcl to instead look like
```hcl
resource "snowflake_role_grants" "this_works" {
  role_name = snowflake_role.foo.name
  roles     = ["roleA"]
  users = [
    "userA",
    "userB",
    "userC",
  ]
}
```
Then `data.Get("users")` does give me back `[userA, userB, userC]`.
Interestingly enough, this happens when the shema for `"users"` is either a `TypeSet` or a `TypeList`.
A pull request I was working on https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/91/files#r333768564 when I found this issue.

**Comment by jbardin** [comment]
Hi @edulop91,

What I think is happening is that terraform can't find the resource that doesn't exist, but isn't able to definitively error out at that point, so the evaluation ends up with an unknown value being sent to the provider. 

This is an instance that will be fixed by #22846. Can you test the issue with the current master branch?

**Comment by jbardin** [comment]
This specific instance is closed by #22846. We can open a new issue if there is more information.
